chainstate/src/detail/chainstateref.rs

```diff
@@ -528,25 +530,35 @@ impl<'a, S: BlockchainStorageRead, O: OrphanBlocks> ChainstateRef<'a, S, O> {
         let mut tx_verifier =
             TransactionVerifier::new(&self.db_tx, utxo_view.derive_cache(), self.chain_config);
 
-        tx_verifier.connect_transaction(
+        let reward_fees = tx_verifier.connect_transactable(
             BlockTransactableRef::BlockReward(block),
             spend_height,
             &median_time_past,
             blockreward_maturity,
         )?;
+        debug_assert!(reward_fees.is_none());
 
         // TODO: add a test that checks the order in which txs are connected
-        for (tx_num, _tx) in block.transactions().iter().enumerate() {
-            tx_verifier.connect_transaction(
-                BlockTransactableRef::Transaction(block, tx_num),
-                spend_height,
-                &median_time_past,
-                blockreward_maturity,
-            )?;
-        }
+        let total_fees = block
+            .transactions()
+            .iter()
+            .enumerate()
+            .map(|(tx_num, _)| {
+                tx_verifier.connect_transactable(
+                    BlockTransactableRef::Transaction(block, tx_num),
+                    spend_height,
+                    &median_time_past,
+                    blockreward_maturity,
+                )
+            })
+            .try_fold(Amount::from_atoms(0), |total, fee| {
+                (total + fee?.expect("connect tx should return fees").0).ok_or_else(|| {
+                    ConnectTransactionError::FailedToAddAllFeesOfBlock(block.get_id())
+                })
+            })?;
```

Comment:
Would it make sense to fuse the `map` and `try_fold` calls?

I.e. replace this pattern:
```rust
iter.map(|a| f(a)).try_fold(b0, |b, fa| g(b, fa))
```
with
```rust
iter.try_fold(b0, |b, a| g(b, f(a)))
```